.github/workflows/update-dependabot-pr.yml

```diff
@@ -0,0 +1,41 @@
+---
+# Dependabot Pull Requests are not updating the NOTICE.txt file, which causes the lint job to fail.
+# This workflow will checkout the dependabot PR, update the NOTICE.txt file, and push the changes back to the PR.
+name: update-dependabot-pr
+
+on:
+  workflow_dispatch: ~ # This gives us the ability to run it for existing PRs where dependabot has already created a PR
+  push:
+    branches:
+      - dependabot/go_modules/**
+
+permissions:
+  contents: read
+
+jobs:
+  update-dependbot-pr:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: elastic/apm-pipeline-library/.github/actions/github-token@current
```

Comment:
@v1v 

I was also thinking.. maybe we could use the github actions token instead with

```
permissions:
  contents: write
```

But then, I guess, the tests won't run.

WDYT?

Reply:
Oh wait. Maybe we actually want to use the github actions token.

pushing to the branch will cause the workflow to be triggered again and again.

Reply:
In addition, using the `GitHub bot` will prevent the build from being rerun.

However, I wonder whether a changeset validation should be in place, so if changes other than `NOTICE`, then run the next steps.

What do you think?

Reply:
Aha, 

we could do 

```
    paths-ignore:
      - NOTICE.txt
```

Then this check will only run if the `NOTICE.txt` file was not changed. Hence, this workflow won't run twice.